torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -933,21 +934,64 @@ def _run_param_init_fn():
         self._hook_registered = False
 
     def _init_param_exec_order_wrap_policy(self, *args, **kwargs) -> None:
+        auto_wrap_policy = kwargs["auto_wrap_policy"]
+        module = kwargs["module"]
+        if isinstance(
+            auto_wrap_policy.tracing_config,
+            TracingConfig
+        ):
+            tracer = auto_wrap_policy.tracing_config.tracer
+            execution_info = _init_execution_info(module)
+            with _patch_tracer(
+                tracer=tracer,
+                root_module=module,
+                execution_info=execution_info,
+            ):
+                try:
+                    tracer.trace(module, auto_wrap_policy.tracing_config.concrete_args)
+                except BaseException as e:
+                    raise RuntimeError(
+                        "tracer.trace failed inside _init_param_exec_order_wrap_policy"
```

Comment:
In what cases might the FX provided tracer be unable to trace through a module's forward graph?

Reply:
The two cases below may be unable to trace:
1. when there is a data-dependent branch
2. when the forward pass contains operators that don't support `torch.fx.Proxy` as the input type (e.g. "arange", "zeros", "ones", "full", "full_like", "eye", "empty", "tensor") (to fix that, users can patch these functions to get them working, for example https://github.com/huggingface/transformers/blob/main/src/transformers/utils/fx.py#L645)